src/Model/Command/Interaction.php

```diff
@@ -121,16 +84,53 @@ public function getCommandType(): string
         return 'interaction';
     }
 
+    /**
+     * Set all Interaction attributes. All previously set attributes are removed.
+     */
+    public function setAttributes(array $attributes): self
+    {
+        $this->attributes = new ArrayObject();
+        foreach ($attributes as $name => $value) {
+            $this->setAttribute($name, $value);
+        }
+
+        return $this;
+    }
+
+    /**
+     * Set Interaction attribute and its value. If attribute with the same name
+     * already exists, it's replaced.
+     *
+     * @param mixed $value
+     */
+    public function setAttribute(string $name, $value): self
+    {
+        Assertion::typeIdentifier($name);
+
+        $this->attributes[$name] = $value;
+
+        return $this;
+    }
+
     public function getCommandParameters(): array
     {
-        return [
-            'interaction_type' => $this->interactionType->jsonSerialize(),
+        $common = [
+            'interaction_type' => $this->interactionType,
             'user_id' => $this->userId,
-            'item_id' => $this->itemId,
             'timestamp' => $this->timestamp,
-            'value' => $this->value,
-            'context' => $this->context,
+            'attributes' => $this->attributes,
+            $this->itemIdAlias => $this->itemId,
         ];
+        $common[$this->itemIdAlias] = $this->itemId;
```

Comment:
Isn't this redundant? It is already set two lines above.

Reply:
Yes, relic from refactoring. Just to explain, the original plan was that there could be multiple aliased ids.